src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -539,9 +540,30 @@ public void validateForStudentExamGeneration(Exam exam) throws BadRequestAlertEx
         List<StudentDTO> notFoundStudentsDtos = new ArrayList<>();
         for (var studentDto : studentDtos) {
             var registrationNumber = studentDto.getRegistrationNumber();
+
+            // If no registration number is given try to fetch it via the login
+            if (!StringUtils.hasText(registrationNumber)) {
+                var userOptional = userService.getUserByLogin(studentDto.getLogin());
+                if (userOptional.isEmpty()) {
+                    notFoundStudentsDtos.add(studentDto);
+                    continue;
+                }
+
+                registrationNumber = userOptional.get().getRegistrationNumber();
+            }
+
             try {
                 // 1) we use the registration number and try to find the student in the Artemis user database
                 Optional<User> optionalStudent = userService.findUserWithGroupsAndAuthoritiesByRegistrationNumber(registrationNumber);
+                if (optionalStudent.isEmpty()) {
+                    // An invalid registration number might be corrected by getting the number from the login
+                    var userOptional = userService.getUserByLogin(studentDto.getLogin());
+                    if (userOptional.isPresent()) {
+                        registrationNumber = userOptional.get().getRegistrationNumber();
+                        optionalStudent = userService.findUserWithGroupsAndAuthoritiesByRegistrationNumber(registrationNumber);
+                    }
```

Comment:
To clarify as I did not get this from the PR description: Logic is number first, login as a fallback? What happens if both are specified and do not match?

Reply:
Yes, the login is just a fallback in case the number is invalid or empty. The number takes precedence if both are present.